### PR TITLE
Added TDX V1.5 quote compatibility

### PR DESCRIPTION
**Pull request by justanotherminh**
Updated the codebase to handle TDX quote version 1.5. Added test cases for this new version.

### CODE REVIEW

**Comment by mingweishih** [comment]
bors try

File: common/tdx/verifier.c [hunk 1]
```diff
@@ -150,8 +152,22 @@ static oe_result_t _fill_with_known_tdx_claims(
...
     tdx_quote = (tdx_quote_t*)quote;
-    tdx_report = &tdx_quote->report_body;
+    if (tdx_quote->version == 4)
+    {
+        tdx_report = &tdx_quote->report_body;
+    }
+    else if (tdx_quote->version == 5)
+    {
+        tdx_quote_v5 = (tdx_quote_v5_t*)quote;
+        if (tdx_quote_v5->type == 1)
```
Comment (mingweishih) [inline common/tdx/verifier.c:164]: Can you add comment on what does each type mean?

File: common/tdx/verifier.c [hunk 2]
```diff
@@ -150,8 +152,22 @@ static oe_result_t _fill_with_known_tdx_claims(
...
+            OE_RAISE(OE_UNEXPECTED);
+
+        tdx_report = (tdx_report_body_t*)&(tdx_quote_v5->body[0]);
+        if (tdx_quote_v5->type == 3)
+            tdx_report_v5 = (tdx_report_body_v5_t*)&(tdx_quote_v5->body[0]);
```
Comment (mingweishih) [inline common/tdx/verifier.c:169]: Same here

**Review by mingweishih — COMMENTED** [review]
Can you also add simple tests on verifying if the newly added claims present for v5 quote?

File: tests/secure_verify/CMakeLists.txt [hunk 3]
```diff
@@ -66,12 +66,12 @@ if (BUILD_ENCLAVES)
   add_test(
     NAME tests/secure_verify_tdx_openssl
     COMMAND $<TARGET_FILE:secure_verify> secure_verify_enc_openssl.signed
-            ${CMAKE_CURRENT_SOURCE_DIR}/data/tdx_quote -f tdx_quote
+            ${CMAKE_CURRENT_SOURCE_DIR}/data/tdx_quote_v5 -f tdx_quote_v5
```
Comment (asvrada) [inline tests/secure_verify/CMakeLists.txt:69]: Could you add test cases for both v4 and v5 quote?

**Comment by asvrada** [comment]
Could you add links here for the source of the definition of quote v5? Why `tdx_quote_v5_t` and `tdx_report_body_v5_t` are defined like so.

File: tests/host_verify/host/CMakeLists.txt [hunk 4]
```diff
@@ -12,5 +12,9 @@ configure_file(../data/sgx_report_bad.bin
                ${CMAKE_CURRENT_BINARY_DIR}/sgx_report_bad.bin COPYONLY)
 configure_file(../data/tdx_quote.bin ${CMAKE_CURRENT_BINARY_DIR}/tdx_quote.bin
                COPYONLY)
+configure_file(../data/quote_v5_Mig.dat ${CMAKE_CURRENT_BINARY_DIR}/quote_v5_Mig.dat
```
[the remark this answers is not in the document]
Comment (justanotherminh) [reply tests/host_verify/host/CMakeLists.txt]: done

File: common/tdx/verifier.c (hunk 1 above)
Comment (justanotherminh) [reply common/tdx/verifier.c:167]: added

**Comment by justanotherminh** [comment]
bors try

**Comment by justanotherminh** [comment]
> Could you add links here for the source of the definition of quote v5? Why `tdx_quote_v5_t` and `tdx_report_body_v5_t` are defined like so.

@asvrada is this what you're looking for? https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/master/QuoteGeneration/quote_wrapper/common/inc/sgx_quote_5.h

**Review by asvrada — APPROVED** [review]

File: common/tdx/verifier.c [hunk 5]
```diff
@@ -150,8 +152,26 @@ static oe_result_t _fill_with_known_tdx_claims(
...
+     * number */
     tdx_quote = (tdx_quote_t*)quote;
-    tdx_report = &tdx_quote->report_body;
+    if (tdx_quote->version == 4)
+    {
+        tdx_report = &tdx_quote->report_body;
+    }
+    else if (tdx_quote->version == 5)
+    {
+        /* If quote version is 5, then recast to TDX quote v5 struct*/
```
Comment (mingweishih) [inline common/tdx/verifier.c:164]:
```suggestion
        /* If quote version is 5, then recast to TDX quote v5 struct */
```

**Review by mingweishih — COMMENTED** [review]
It would be ideal if we can add a simple test that validates the v5 quote does result in 2 additional claims.

**Comment by justanotherminh** [comment]
> It would be ideal if we can add a simple test that validates the v5 quote does result in 2 additional claims.

@mingweishih I added a small check for this in `host_verify\host\host.cpp`. Can you check to see if that's what you wanted?

**Review by mingweishih — APPROVED** [review]

**Comment by mingweishih** [comment]
bors try

**Comment by mingweishih** [comment]
bors try

**Comment by justanotherminh** [comment]
bors try

**Pull request merged**